Framework/Geometry/test/XMLInstrumentParameterTest.h

```diff
@@ -50,42 +51,48 @@ class XMLInstrumentParameterTest : public CxxTest::TestSuite {
 
 public:
   void test_throws_with_unknown_flag() {
+    Kernel::TimeROI *roi = nullptr;
     TimeSeriesProperty<double> series("doubleProperty");
     series.addValue("2000-11-30T01:01:01", 1);
 
     const std::string made_up_flag = "mode"; // We do not support mode statistics filtering.
     XMLInstrumentParameter_sptr logFile = make_logfile_object(made_up_flag);
 
-    TSM_ASSERT_THROWS("Unknown flag should cause failure", logFile->createParamValue(&series),
+    TSM_ASSERT_THROWS("Unknown flag should cause failure", logFile->createParamValue(&series, roi),
                       const Kernel::Exception::InstrumentDefinitionError &)
   }
 
   void test_filter_by_first_value() {
+    Kernel::TimeROI *roi = nullptr;
     TimeSeriesProperty<double> series("doubleProperty");
 
     const double expectedFilteredValue = 1;
     series.addValue("2000-11-30T01:01:01", expectedFilteredValue);
     series.addValue("2000-11-30T01:01:02", 2);
 
     XMLInstrumentParameter_sptr logFile = make_logfile_object("first_value");
-    const double actualFilteredValue = logFile->createParamValue(&series);
+    const double actualFilteredValue = logFile->createParamValue(&series, roi);
     TSM_ASSERT_EQUALS("Filtering by First Value is not performed correctly", expectedFilteredValue,
                       actualFilteredValue);
   }
 
   void test_filter_by_last_value() {
+    Kernel::TimeROI *roi = new Kernel::TimeROI;
+    roi->addROI("2000-11-30T01:01:01", "2000-11-30T01:01:02");
     TimeSeriesProperty<double> series("doubleProperty");
 
     const double expectedFilteredValue = 1;
     series.addValue("2000-11-30T01:01:01", 0);
     series.addValue("2000-11-30T01:01:02", expectedFilteredValue);
 
     XMLInstrumentParameter_sptr logFile = make_logfile_object("last_value");
-    const double actualFilteredValue = logFile->createParamValue(&series);
+    const double actualFilteredValue = logFile->createParamValue(&series, roi);
```

Comment:
can you add additional test with different ROI's?

roi->addROI("2000-11-30T01:01:00", "2000-11-30T01:01:01");
In this case, actualFilteredValue should be 0